rfc-113-expanding-draft-access-for-unauthenticated-users.md

```diff
@@ -0,0 +1,212 @@
+  edition through link expansion rules.
+- Publishing API and Content Store will have their access limit code decoupled
+  from auth-bypass ids. This will be done to remove a frequent source of
+  confusion that conflates the separate processes of access limiting and
+  auth-bypass.
+- Collections Publisher will be changed to include a flag in auth-bypass tokens
+  that provide draft asset access. To reflect the increased access of a token
+  they will be given an expiry time (anticipated to be 1 month). Collections
+  Publisher will also be updated to include origin information in the token
+  (such as creator id and content id) to allow a token to be audited. The
+  [auth-bypass token documentation][auth-bypass-token-docs] will be updated to
+  reflect this as the best practice for future auth-bypass tokens
+
+### Rejected alternative approaches
+
+There were a number of alternative ideas which were considered to provide fact
+checking for step-by-steps and were ultimately rejected. This is a quick
+compilation of them and the reasons why they were rejected.
+
+#### Providing a link to preview the embedded part of a step-by-step distinct from the content
+
+This would involve creating a page which shows just the step-by-step
+navigation component that is embedded on linked pages. This was deemed
+insufficient as fact checkers are often comparing the step-by-step navigation
+with the context of the specific guidance pages.
+
+#### Providing a means to embed the step-by-step component into live content
+
+This would involve a means of injecting the step-by-step navigation component
+into live GOV.UK pages via JavaScript. This would prevent a fact checker of a
+step-by-step from viewing any draft content except the step-by-step
+navigation itself. This was quickly rejected as it was considered to be a
+common scenario for the pages that are part of a step-by-step
+to also be drafts.
+
+#### Exclude asset access from fact-checking
+
+Assets are the area of access that provide the most complication since they
+are distinct from the Publishing API link system. An option was to exclude
+them from the access checks and have them return forbidden errors if a
+step-by-step page used them - this would typically be most noticeable on an
+embedded image.
+
+This option was rejected due to it leaving a problem that could be easily
+anticipated and likely having to fall to a support team to resolve or explain
+whenever it occurs.
+
+#### Providing a fact checker access to all resources on the draft stack that are not access limited
+
+An approach that was considered was to allow the fact checker to have access
+to all pages on the draft stack that are not access limited. This would grant
+them similar access to an authenticated user. This offered the advantage that
+it allows simplifying the logic involved in checking tokens and is consistent
+with privacy expectations.
+
+This was rejected since it was a significant departure from the existing access
```

Comment:
> This is interesting, since we seem to be taking this approach with assets. Perhaps that's less of a concern, since the primary route to an asset would be through a document i.e. unrelated assets are effectively hidden if we scope draft access to only the content of the step-by-step. 

Yes, the route to view an asset would be through having to know what both the id and the filename were for that asset. Whereas for content the route would be to browse the site and/or guess slugs (but still be restricted by search not existing for draft content). 

In practice the proposed change of this RFC would be very hard for any fact-checkers to be aware that there is a difference based on this flag as viewing content would be no different and for applications that don't set the flag there'd be no change. Whereas for this rejected change it'd be very easy for fact checkers to experience a difference in behaviour by just clicking a link. 

I actually think that it should be ok for fact-checkers to view the whole draft stack and I imagine if GOV.UK increases it's journey into embedding content from multiple places (as expected with more embracing of structured content) we'll have more questions to resolve on how that is fact checked and what the meaning of access limiting is. However that is quite a big change and could be contentious/surprising so, given the step-by-step team is to exist for only another couple of weeks, it is not a conversation I want to start unnecessarily.

> Can we address/discuss this same issue for assets as part of the proposal?

It's referenced in the consequences section.